frame/staking/src/tests.rs

```diff
@@ -513,29 +511,29 @@ fn nominating_and_rewards_should_work() {
 			}
 
 			// bond two account pairs and state interest in nomination.
-			// 2 will nominate for 10, 20, 30
 			assert_ok!(Staking::bond(Origin::signed(1), 2, 1000, RewardDestination::Controller));
 			assert_ok!(Staking::nominate(Origin::signed(2), vec![11, 21, 31]));
-			// 4 will nominate for 10, 20, 40
+
 			assert_ok!(Staking::bond(Origin::signed(3), 4, 1000, RewardDestination::Controller));
 			assert_ok!(Staking::nominate(Origin::signed(4), vec![11, 21, 41]));
 
 			// the total reward for era 0
 			let total_payout_0 = current_total_payout_for_duration(reward_time_per_era());
 			<Pallet<Test>>::reward_by_ids(vec![(41, 1)]);
-			<Pallet<Test>>::reward_by_ids(vec![(31, 1)]);
+			<Pallet<Test>>::reward_by_ids(vec![(21, 1)]);
 
 			mock::start_active_era(1);
 
 			// 10 and 20 have more votes, they will be chosen.
 			assert_eq_uvec!(validator_controllers(), vec![20, 10]);
 
-			// OLD validators must have already received some rewards.
+			// old validators must have already received some rewards.
+			let initial_balance_40 = Balances::total_balance(&40);
+			let mut initial_balance_20 = Balances::total_balance(&20);
 			mock::make_all_reward_payment(0);
-			assert_eq!(Balances::total_balance(&40), 1 + total_payout_0 / 2);
-			assert_eq!(Balances::total_balance(&30), 1 + total_payout_0 / 2);
-
-			// ------ check the staked value of all parties.
+			assert_eq!(Balances::total_balance(&40), initial_balance_40 + total_payout_0 / 2);
+			assert_eq!(Balances::total_balance(&20), initial_balance_20 + total_payout_0 / 2);
+			initial_balance_20 = Balances::total_balance(&20);
 
 			// 30 and 40 are not chosen anymore
 			assert_eq!(
```

Comment:
beyond the scope of this PR but could use `mock::active_era()` below instead of `Staking::active_era().unwrap().index`

Reply:
replaced all.